sdcm/utils/replication_strategy_utils.py

```diff
@@ -225,8 +225,8 @@ def decrease_keyspaces_rf(self):
             replication-factor value.
         """
         node = self.target_node
-        # Ensure that nodes_num is 2 or greater
-        if self.original_nodes_number > 1:
+
+        if self.original_nodes_number > 0:
```

Comment:
1. This won't work in case of a single DC with one node.
2. It would be a problem to remove a one-node DC if its keyspaces are not replicated in other DCs. It would probably fail altering keyspace.
3. That's the reason the for original condition request.
So it could be possible to have an additional check that other DCs exist and they replicate all keyspaces that are about to be removed from the dc-to-remove.
But it think it must be differentiated and much clearer as a different scenario and not just a simple node decommission.
IMO SCT better add a function like `remove_dc` which will take care and report for this special process of removing a whole DC, not just a node. of course it can then use this code as well.

Reply:
`remove_dc` is a great idea as we are currently trying to do that with tools (methods) that are not designed to do that.